core/src/main/scala/org/http4s/headers/Age.scala

```diff
@@ -1,5 +1,31 @@
 package org.http4s
 package headers
 
-object Age extends HeaderKey.Default
+import org.http4s.util.{Renderer, Writer}
+import org.http4s.util.Renderable._
+import org.http4s.parser.HttpHeaderParser
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration._
+
+object Age extends HeaderKey.Internal[Age] with HeaderKey.Singleton {
+  override def parse(s: String): ParseResult[Age] =
+    HttpHeaderParser.AGE(s)
+}
+
+/**
+  * Constructs an Age header.
+  *
+  * The value of this field is a positive number of seconds (in decimal) with an estimate of the amount of time since the response
+  *
+  * @param age age of the response
+  */
+final case class Age(age: FiniteDuration) extends Header.Parsed {
+  require(age >= 0.seconds)
```

Comment:
No, it would be on the `Age` companion object, as an alternative to `apply`.

I think the standard could look something like this:
- all constructors are total unless prefixed with `unsafe`
- because that's a pain, `unsafe` constructors are provided for people who don't mind throwing
- macro versions that support literals without a `ParseResult` wrapper if we can validate them at compile time.  Since there aren't literal durations, the macro is a little harder to provide here.

Another option, to avoid `unsafe` when 99% of people will do it right, is to treat all negative values as duration zero.  But I don't think I support that.